### DISCUSSION TITLE
Memory and time to generate spreadsheet

### DISCUSSION THREAD

**Discussion by dbuerer** [PHPOffice/PhpSpreadsheet#1868] [Q&A]
I used PHPExcel for years with no problems...then migrated to PHPSpreadsheet due to the upgrade of PHP Version.  I mostly using it for reading/importing spreadsheet data, and, filling out spreadsheet templates and sending them to our users.  

Regarding one of my templates....

1. With PHP Excel it took about 10-15 seconds to generate the spreadsheet, and, it never failed

2. With PHP Spreadsheet, I get the out of memory error all the time.  I'm successful about 20% of the time and the other 80% i get the out of memory error. This is on a workbook that has about eight spreadsheets with 100 rows each.  It seems to be very inconsistent.

3.  I've implemented using Symfony PSR cache using an SQL server for for the cache files, and, i don't know whether it ever will actually complete.  I've given up at 30 minutes about and tens of thousands of cached cells in the database.

How do i fix this?

**Comment by pgrimaud** [comment]
Did you try Symfony PSR cache using Redis?

**Reply by dbuerer** [reply]
only because you suggested it have i now tried it.  Thank you for the good idea.  Kind of a pain to get setup, however, with this cache it at least completes the spreadsheet (eventually.)  I have a problem with it timing out around 12 minutes when attempting 136 cells across 7 worksheets, and save it.  With PHP excel it would've taken about 15 seconds.
